src/server/game/AI/CreatureAI.cpp

```diff
@@ -271,6 +271,112 @@ bool CreatureAI::_EnterEvadeMode()
     return true;
 }
 
+#define BOUNDARY_VISUALIZE_CREATURE 15425
+#define BOUNDARY_VISUALIZE_CREATURE_SCALE 0.25f
+#define BOUNDARY_VISUALIZE_STEP_SIZE 1
+#define BOUNDARY_VISUALIZE_FAILSAFE_LIMIT 200
+#define BOUNDARY_VISUALIZE_SPAWN_HEIGHT 5
+void CreatureAI::VisualizeBoundary(uint32 duration, Unit* owner, bool fill) const
+{
+    typedef std::pair<int32, int32> coordinate;
+    
+    if (!owner)
+        return;
+
+    if (!_boundary || _boundary->empty())
+    {
+        me->Talk("VisualizeBoundary error: Creature movement is unbounded", CHAT_MSG_MONSTER_YELL, LANG_UNIVERSAL, 50000.0f, nullptr);
+        return;
+    }
+
+    std::queue<coordinate> Q;
+    std::unordered_set<coordinate> alreadyChecked;
+    std::unordered_set<coordinate> outOfBounds;
+
+    float meX = me->GetPositionX(), meY = me->GetPositionY(), meZ = me->GetPositionZ(), spawnZ = meZ + BOUNDARY_VISUALIZE_SPAWN_HEIGHT;
+    
+    if (!CheckBoundary(me))
+    {
+        me->Talk("VisualizeBoundary warning: Current position is out of bounds", CHAT_MSG_MONSTER_YELL, LANG_UNIVERSAL, 50000.0f, nullptr);
+        return;
+    }
+    bool hadBoundsWarning = false;
+    Q.push({ 0,0 });
+    while (!Q.empty())
+    {
+        coordinate front = Q.front();
+        bool hasOutOfBoundsNeighbor = false;
+        for (coordinate off : std::list<coordinate>({ {1, 0}, { 0,1 }, { -1,0 }, { 0,-1 } }))
```

Comment:
Or you could use `std::initializer_list`  instead of `std::list` [which the compiler will optimize out:
](https://goo.gl/DqxFGt)
```c++
for (int i : std::initializer_list<int>{1, 2, 3, 4})
{
    // ...
}
```